sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -320,8 +321,14 @@ abstract class UnaryNode extends LogicalPlan {
   override def statistics: Statistics = {
     // There should be some overhead in Row object, the size should not be zero when there is
     // no columns, this help to prevent divide-by-zero error.
-    val childRowSize = child.output.map(_.dataType.defaultSize).sum + 8
-    val outputRowSize = output.map(_.dataType.defaultSize).sum + 8
+    val childRowSize = child.output.map(n => n.dataType match {
```

Comment:
Nit, but this can be simpler in Scala

```
val childRowSize = child.output.map(_.dataType).map {
  case _: ObjectType => 0L
  case dt => dt.defaultSize
}.sum + 8
```